File: backend/pkg/models/post.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Post struct {
	ID            uuid.UUID `json:"id"`
	AuthorID      uuid.UUID `json:"author_id"`
	VoxsphereID   uuid.UUID `json:"voxsphere_id"`
	Title         string    `json:"title"`
	Text          string    `json:"text"`
	TextHtml      string    `json:"text_html"`
	Ups           int32     `json:"ups"`
	Over18        bool      `json:"over18"`
	Spoiler       bool      `json:"spoiler"`
	CreatedAt     time.Time `json:"created_at"`
	CreatedAtUnix int64     `json:"created_at_unix"`
	UpdatedAt     time.Time `json:"updated_at"`
}

type PostPaginated struct {
	ID            uuid.UUID `json:"id"`
	Author        string    `json:"author"`
	Voxsphere     string    `json:"voxsphere"`
	AuthorID      uuid.UUID `json:"author_id"`
	VoxsphereID   uuid.UUID `json:"voxsphere_id"`
	Title         string    `json:"title"`
	Text          string    `json:"text"`
	TextHtml      string    `json:"text_html"`
	MediaType     MediaType `json:"media_type"`
	Medias        []any     `json:"medias"`
	Ups           int32     `json:"ups"`
	NumComments   int32     `json:"num_comments"`
	NumAwards     int32     `json:"num_awards"`
	Over18        bool      `json:"over18"`
	Spoiler       bool      `json:"spoiler"`
	CreatedAt     time.Time `json:"created_at"`
	CreatedAtUnix int64     `json:"created_at_unix"`
	UpdatedAt     time.Time `json:"updated_at"`
}

File: frontend/src/post/post.ts
export enum MediaType {
    Image = 'image',
    Gif = 'gif',
    Video = 'video',
    Gallery = 'gallery',
    Link = 'link',
    Multi = 'multi',
    Text = 'text'
}

export interface ImageMetadata {
    id: string;
    image_id: string;
    height: number;
    width: number;
    url: string;
    created_at: Date;
    created_at_unix: number;
    updated_at: Date;
}

export interface GifMetadata {
    id: string;
    gif_id: string;
    height: number;
    width: number;
    url: string;
    created_at: Date;
    created_at_unix: number;
    updated_at: Date;
}

export interface GalleryMetadata {
    id: string;
    gallery_id: string;
    order_index: number;
    height: number;
    width: number;
    url: string;
    created_at: Date;
    created_at_unix: number;
    updated_at: Date;
}

export interface Video {
    id: string;
    media_id: string;
    url: string;
    height: number;
    width: number;
    created_at: Date;
    created_at_unix: number;
    updated_at: Date;
}

export interface LinkType {
    id: string;
    media_id: string;
    link: string;
    image: ImageMetadata[];
    created_at: Date;
    created_at_unix: number;
    updated_at: Date;
}

type PostMedia =
    | ImageMetadata
    | GifMetadata
    | GalleryMetadata
    | Video
    | LinkType;

export interface Post {
    id: string;
    author: string;
    author_id: string;
    voxsphere: string;
    voxsphere_id: string;
    title: string;
    text: string;
    text_html: string;
    media_type: MediaType;
    medias: PostMedia[];
    ups: number;
    num_comments: number;
    num_awards: number;
    over18: boolean;
    spoiler: boolean;
    created_at: Date;
    created_at_unix: number;
    updated_at: Date;
}
